client/components/jetpack-header/index.jsx

```diff
@@ -36,25 +34,6 @@ export class JetpackHeader extends PureComponent {
 			return null;
 		}
 
-		if ( isWooOnboarding ) {
```

Comment:
cc @mirka in case this deletion unlocks further improvements to the `JetpackHeader` component